lambdas/build-proven-user-identity-details/src/main/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandler.java

```diff
@@ -150,46 +149,44 @@ public Map<String, Object> handleRequest(JourneyRequest input, Context context)
             List<VcStoreItem> credentialIssuerItems, List<VcStatusDto> currentVcStatuses)
-            throws ParseException, JsonProcessingException, ProvenUserIdentityDetailsException,
-                    NoVcStatusForIssuerException {
-        for (VcStoreItem item : credentialIssuerItems) {
-            if (EVIDENCE_CRI_TYPES.contains(item.getCredentialIssuer())
+            throws ProvenUserIdentityDetailsException, NoVcStatusForIssuerException,
+                    CredentialParseException {
+        final List<VcStoreItem> successfulVCStoreItems = new ArrayList<>();
+        for (VcStoreItem credentialIssuerItem : credentialIssuerItems) {
+            if (userIdentityService.isEvidenceVc(credentialIssuerItem)
                     && userIdentityService.isVcSuccessful(
                             currentVcStatuses,
-                            configService.getComponentId(item.getCredentialIssuer()))) {
-                JsonNode vcSubjectNode =
-                        mapper.readTree(
-                                        SignedJWT.parse(item.getCredential())
-                                                .getPayload()
-                                                .toString())
-                                .path(VC_CLAIM)
-                                .path(VC_CREDENTIAL_SUBJECT);
-
-                JsonNode nameNode = vcSubjectNode.path(NAME_PROPERTY_NAME);
-
-                Name name = mapper.convertValue(nameNode.get(0), Name.class);
-
-                StringBuilder nameBuilder = new StringBuilder();
-                name.getNameParts()
-                        .forEach(
-                                namePart -> {
-                                    if (nameBuilder.length() == 0) {
-                                        nameBuilder.append(namePart.getValue());
-                                    } else {
-                                        nameBuilder.append(" ").append(namePart.getValue());
-                                    }
-                                });
-
-                JsonNode dateOfBirthNode = vcSubjectNode.path(BIRTH_DATE_PROPERTY_NAME);
-
-                BirthDate birthDate = mapper.convertValue(dateOfBirthNode.get(0), BirthDate.class);
-
-                return new NameAndDateOfBirth(nameBuilder.toString(), birthDate.getValue());
+                            configService.getComponentId(
+                                    credentialIssuerItem.getCredentialIssuer()))) {
+                successfulVCStoreItems.add(credentialIssuerItem);
             }
         }
-        LOGGER.error("Failed to find name and date of birth of proven user identity");
-        throw new ProvenUserIdentityDetailsException(
-                "Failed to find name and date of birth of proven user identity");
+
+        try {
+            final IdentityClaim identityClaim =
+                    userIdentityService.findIdentityClaim(successfulVCStoreItems);
```

Comment:
The `succesfulVCStoreItems` list here has been filtered by calling `isEvidenceVc()` on items before adding. That's happening again within `findIdentityClaim()`. You might be able to simplify the above by not doing the first filter?

Reply:
Good point - I've moved out the `isEvidenceVc()` such that `findIdentityClaim` is as method that just accepts a list of successful evidence vcs